ksql-cli/src/main/java/io/confluent/ksql/cli/Cli.java

```diff
@@ -150,60 +158,19 @@ private void displayWelcomeMessage() {
     terminal.flush();
   }
 
-  public void runNonInteractively(final String input) throws Exception {
-    // Allow exceptions to halt execution of the Ksql script as soon as the first one is encountered
-    for (final String logicalLine : getLogicalLines(input)) {
-      try {
-        handleLine(logicalLine);
-      } catch (final EndOfFileException exception) {
-        // Swallow these silently; they're thrown by the exit command to terminate the REPL
-        return;
-      }
-    }
-  }
-
-  @SuppressWarnings("RedundantStringOperation") // Incorrect warning. Operation is not redundant
-  private List<String> getLogicalLines(final String input) {
-    final List<String> result = new ArrayList<>();
-    final StringBuilder logicalLine = new StringBuilder();
-
-    Arrays.stream(input.split("\n"))
-        .map(String::trim)
-        .filter(line -> !line.isEmpty())
-        .forEach(physicalLine -> {
-          if (physicalLine.endsWith("\\")) {
-            logicalLine.append(physicalLine.substring(0, physicalLine.length() - 1));
-          } else {
-            result.add(logicalLine.append(physicalLine).toString().trim());
-            logicalLine.setLength(0);
-          }
-        });
-
-    return result;
-  }
-
   @Override
-  public void close() throws IOException {
+  public void close() {
     queryStreamExecutorService.shutdownNow();
-    restClient.close();
```

Comment:
Any specific reason to remove this line?

Reply:
Because the Cli does not own the rest client. It didn't create it, so should not close it.

Reply:
Line 51 of Ksql.java now wraps the rest client in a try-with-resource block:

```
try (KsqlRestClient restClient = new KsqlRestClient(options.getServer(), properties)) {
```

Meaning the class that creates the resource is now responsible for its lifecycle, which is a good pattern to follow.